### PR TITLE
docs: add FAQ and best practice docs

### PR DESCRIPTION
**Pull request by cqqqq777**
#### Check the PR title.

- [ ] This PR title match the format: \<type\>(optional scope): \<description\>
- [ ] The description of this PR title is user-oriented and clear enough for others to understand.

#### (Optional) More detailed description for this PR(en: English/zh: Chinese).

en:
zh(optional):

### CODE REVIEW

File: content/zh/docs/kitex/Best Practice/self_check/panic_self_check.md [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+遇到任何 panic，都请按以下 4 步原则来排查。
+
+1. panic 的原因
+2. 栈从哪开始看
+3. 报错的方法是什么
+4. 报错具体代码位置在哪 
+
+下图是一个典型的 panic 错误栈，我们在图中描述了四项主要信息的位置。 
+
+![image](/img/blog/Kitex_self_check/image.png)
```
Comment (GuangmingLuo) [inline content/zh/docs/kitex/Best Practice/self_check/panic_self_check.md:21]: 这个文件的所有图片名字有问题

File: content/zh/docs/kitex/Best Practice/self_check/panic_self_check.md [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+/.../go/src/runtime/panic.go:\d +0x\d
+```
+
+栈的最顶部分是描述框架捕获了这个 panic，并不是 panic 的原因，属于 recover 后的操作内容，对排查问题意义不大，可以直接忽略。
+
+### 报错的具体方法
+
+找到了错误栈的起始位置后，紧跟着的下一行，就是 panic 报错的具体方法。
+
+![image](/img/blog/Kitex_self_check/panic_stack.png)
```
Comment (GuangmingLuo) [inline content/zh/docs/kitex/Best Practice/self_check/panic_self_check.md:89]: 为啥这里的图片名都是一样？copy paste 吗？

Comment (cqqqq777) [reply content/zh/docs/kitex/Best Practice/self_check/panic_self_check.md:89]: 都是相同的图片

Comment (cqqqq777) [reply content/zh/docs/kitex/Best Practice/self_check/panic_self_check.md:89]: 图片名字的话渲染过后看不到，我就全部统一命名了，后面的路径正确就可以

File: content/zh/docs/kitex/Best Practice/self_check/mem_leak_self_check.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+### 使用 Sonic 解析 Request/Response 的 string 字段并“缓存”对象
+
+典型 Profile：
+
+![image](/img/blog/Kitex_self_check/cache_sonic_obj.png)
+
+典型触发场景：
+
+1. `Request/Response` 中有一个大 `string` 对象，比如常见的 `abtest` 参数(`string` 类型内容为 JSON)
```
Comment (AsterDY) [inline content/zh/docs/kitex/Best Practice/self_check/mem_leak_self_check.md:132]: 这个abtest内场才有，去掉吧

File: content/zh/docs/kitex/Best Practice/self_check/panic_self_check.md [hunk 4]
```diff
@@ -0,0 +1,356 @@
...
+#### 排查示例
+
+1. 以下错误栈，我们快速看到 `*Device` 是 `0x0`，表明 `*Device` 是个空指针。再根据调用栈向下看，`*Device` 是由 `*Log.writeField5` 调用的。
+
+   因为这里是生成代码，**可以得出结论**，生成代码 `*Log` 结构体的第 5 个字段 `*Device` 没有赋值，造成空指针。具体在哪里没有赋值，基于栈向下找就可以，这里不再赘述。
+
+   ```go
+   KITE: panic in processor: runtime error: invalid memory address or nil pointer dereference
+   goroutine 498022546 [running]:
+   .../code.byted.org/kite/kite.(*RpcServer).processRequests.func1(0xc000424550)
```
Comment (AsterDY) [inline content/zh/docs/kitex/Best Practice/self_check/panic_self_check.md:186]: 内部域名，坐一下模糊处理吧

**Comment by AsterDY** [comment]
> **The latest updates on your projects**. Learn more about [Vercel for Git ↗︎](https://vercel.link/github-learn-more)
> 
> Name	Status	Preview	Comments	Updated (UTC)
> **cloudwego**	❌ Failed ([Inspect](https://vercel.com/cloudwego/cloudwego/BZvwY43G6SKVPS5TB9ZySJR5xTdp))			May 1, 2024 4:04pm

预览部署挂了？

**Comment by GuangmingLuo** [comment]
Error: Error building site: "/vercel/path0/content/en/docs/kitex/Tutorials/code-gen/custom_struct_tags.md:7:1": failed to unmarshal YAML: yaml: line 6: did not find expected comment or line break

File: content/en/docs/kitex/Best Practice/self_check/panic_self_check.md [hunk 5]
```diff
@@ -177,8 +177,8 @@ Using the above knowledge, we can quickly determine if a specific pointer is `ni
    ```go
    KITE: panic in processor: runtime error: invalid memory address or nil pointer dereference
    goroutine 498022546 [running]:
-   .../github.com/cloudwego/kitex/kite.(*RpcServer).processRequests.func1(0xc000424550)
-   /.../github.com/cloudwego/kitex/kite/kite_server.go:227 +0xc8
+   .../xxx/kite.(*RpcServer).processRequests.func1(0xc000424550)
+   /.../xxx/kite/kite_server.go:227 +0xc8
```
Comment (GuangmingLuo) [inline content/en/docs/kitex/Best Practice/self_check/panic_self_check.md:181]: 开源不存在 Kite 目录

File: content/en/docs/kitex/Best Practice/self_check/panic_self_check.md [hunk 6]
```diff
@@ -195,26 +195,26 @@ Using the above knowledge, we can quickly determine if a specific pointer is `ni
    ```go
    KITE: processing request error=KE.UNREGISTERED/0 - ?/0: KITC: panic, runtime error: invalid memory address or nil pointer dereference
    goroutine 271 [running]:
-   .../github.com/cloudwego/kitex/kitc.RPCTimeoutMW.func1.1.1(0x2dad3a0, 0xc0009c3260, 0xc000db81a0, 0x0, 0x0, 0xc0009c32c0)
-   /.../github.com/cloudwego/kitex/kitc/middlewares.go:314 +0xcc
+   .../xxx/kitc.RPCTimeoutMW.func1.1.1(0x2dad3a0, 0xc0009c3260, 0xc000db81a0, 0x0, 0x0, 0xc0009c32c0)
+   /.../xxx/kitc/middlewares.go:314 +0xcc
```
Comment (GuangmingLuo) [inline content/en/docs/kitex/Best Practice/self_check/panic_self_check.md:199]: 还有 kitc

**Review by AsterDY — APPROVED** [review]

File: content/zh/docs/kitex/Best Practice/self_check/panic_self_check.md [hunk 7]
```diff
@@ -0,0 +1,356 @@
...
+
+7. **panic: ......**
+
+   错误原因：有些 `panic` 可能是业务自己通过 `panic("xxx")` 自行打印的，那么在这里 `panic:` 后面就会出现自行打印的内容，可以根据这个内容检索代码，就可以找到 `panic` 的位置，进一步排查错误原因
+
+### 栈的起始位置
+
+知道了错误原因之后，接下来就要定位错误的代码位置。首先我们学习如何看错误栈，从哪里开始看。
+
+![image](/img/blog/x_self_check/panic_stack.png)
```
Comment (GuangmingLuo) [inline content/zh/docs/kitex/Best Practice/self_check/panic_self_check.md:75]: /img/blog/Kitex_self_check/panic_stack.png

File: content/zh/docs/kitex/Best Practice/self_check/panic_self_check.md [hunk 8]
```diff
@@ -0,0 +1,356 @@
...
+/.../go/src/runtime/panic.go:\d +0x\d
+```
+
+栈的最顶部分是描述框架捕获了这个 panic，并不是 panic 的原因，属于 recover 后的操作内容，对排查问题意义不大，可以直接忽略。
+
+### 报错的具体方法
+
+找到了错误栈的起始位置后，紧跟着的下一行，就是 panic 报错的具体方法。
+
+![image](/img/blog/x_self_check/panic_stack.png)
```
Comment (GuangmingLuo) [inline content/zh/docs/kitex/Best Practice/self_check/panic_self_check.md:89]: /img/blog/Kitex_self_check/panic_stack.png

**Pull request merged**